src/stream/src/task/stream_manager.rs

```diff
@@ -573,13 +577,31 @@ impl LocalStreamManagerCore {
                     *ENABLE_ASYNC_STACK_TRACE,
                 );
                 let instrumented = monitor.instrument(traced);
-                tokio::spawn(instrumented)
+
+                let memory_usage_metrics = async move {
+                    loop {
+                        TASK_LOCAL_ALLOC.with(|alloc| {
```

Comment:
We can have a WrapperFuture which records memory usage every time it yields. This might be more efficient and deterministic.

Reply:
If there're some CPU-heavy work, we may find the metrics not updated for a while. Just like the stack trace reporter. 🤔

Reply:
But we are using futures::join, which won't update either when there's CPU intensive jobs.